R/plot-population-time-profile.R

```diff
@@ -0,0 +1,132 @@
+#' Concentration time profile plot for population
+#'
+#' @inheritParams plotIndividualTimeProfile
+#' @param quantiles A numerical vector with quantile values (Default: `c(0.05,
+#'   0.50, 0.95)`). In the profile plot, the middle value will be used to draw a
+#'   line, while the lower and upper values will be used to create a ribbon.
+#'
+#' @import tlf
+#'
+#' @family plotting
+#'
+#' @examples
+#'
+#' # TODO: add example
+#'
+#' @export
+plotPopulationTimeProfile <- function(dataCombined,
+                                      defaultPlotConfiguration = NULL,
+                                      quantiles = c(0.05, 0.5, 0.95)) {
+  # validation -----------------------------
+
+  defaultPlotConfiguration <- defaultPlotConfiguration %||% DefaultPlotConfiguration$new()
+  validateIsOfType(dataCombined, "DataCombined")
+  validateIsSameLength(objectCount(dataCombined), 1L) # only single instance is allowed
+  validateIsOfType(defaultPlotConfiguration, "DefaultPlotConfiguration", nullAllowed = FALSE)
+  validateIsNumeric(quantiles, nullAllowed = FALSE)
+  validateIsOfLength(quantiles, 3L)
+
+  # data frames -----------------------------
+
+  df <- dataCombined$toDataFrame()
+
+  # Getting all units on the same scale
+  df <- .unitConverter(df, defaultPlotConfiguration$xUnit, defaultPlotConfiguration$yUnit)
+
+  # Datasets which haven't been assigned to any group will be plotted as a group
+  # on its own. That is, the `group` column entries for them will be their names.
+  df <- .addMissingGroupings(df)
+
+  # `TimeProfilePlotConfiguration` object -----------------------------
+
+  # Create an instance of `TimeProfilePlotConfiguration` class by doing a
+  # one-to-one mapping of internal plot configuration object's public fields
+  individualTimeProfilePlotConfiguration <- .convertGeneralToSpecificPlotConfiguration(
+    data = df,
+    specificPlotConfiguration = tlf::TimeProfilePlotConfiguration$new(),
+    generalPlotConfiguration = defaultPlotConfiguration
+  )
+
+  # plot -----------------------------
+
+  # Which dataset types are present?
+  datasetTypePresent <- .extractPresentDatasetTypes(dataCombined)
```

Comment:
From here on, same comments as for the "individual" function.

Lots of code duplication, I think most of it can be packed in an internal function. Actually only few lines of code differ.